src/montgomery.rs

```diff
@@ -36,285 +56,145 @@ use field::FieldElement;
 use edwards::{EdwardsPoint, CompressedEdwardsY};
 use scalar::Scalar;
 
-// XXX Move these to a common "group" module?  At the same time, we should
-// XXX probably make a `trait Group` once const generics are implemented in
-// XXX Rust. —isis
-//
-// XXX I put these in a `traits` module for now - hdevalence
 use traits::{Identity, ValidityCheck};
 
 use subtle::ConditionallyAssignable;
 use subtle::ConditionallySwappable;
 use subtle::Equal;
 use subtle::Mask;
 
-/// In "Montgomery u" format, as used in X25519, a point `(u,v)` on
-/// the Montgomery curve
-///
-///    v^2 = u * (u^2 + 486662*u + 1)
-///
-/// is represented just by `u`.  Note that we use `(u,v)` instead of
-/// `(x,y)` for Montgomery coordinates to avoid confusion with Edwards
-/// coordinates.  For Montgomery curves, it is possible to compute the
-/// `u`-coordinate of `n(u,v)` just from `n` and `u`, so it is not
-/// necessary to use `v` for a Diffie-Hellman key exchange.
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
-pub struct CompressedMontgomeryU(pub [u8; 32]);
-
-impl CompressedMontgomeryU {
-    /// View this `CompressedMontgomeryU` as an array of bytes.
+/// Holds the \\(u\\)-coordinate of a point on the Montgomery form of
+/// Curve25519 or its twist.
+#[derive(Copy, Clone, Debug)]
+pub struct MontgomeryPoint(pub [u8; 32]);
```

Comment:
Should this be `pub(crate)`?

Reply:
Err, I mean, should this be `pub struct MontgomeryPoint(pub(crate) [u8; 32]);`

Reply:
No, because the `MontgomeryPoint` works similarly to, say, a `CompressedEdwardsY`, in that it's now a typed wrapper around bytes.